README.md

```diff
@@ -305,6 +305,53 @@ kubectl create secret \
 kubectl apply -f tfstate-default-${NAME}.yaml
 ```
 
+### Health Checks
+
+For some resources, it may be useful to perform health checks on them to verify that they are ready to accept connection before the terraform goes into `Ready` state:
+
+```
+# main.tf
+
+output "rdsURL" {
+  value = "mydb.xyz.us-east-1.rds.amazonaws.com"
+}
+
+output "rdsPort" {
+  value = "3306"
+}
+
+output "myappURL" {
+  value = "https://example.com"
+}
+```
+
+```yaml
+apiVersion: infra.contrib.fluxcd.io/v1alpha1
+kind: Terraform
+metadata:
+  name: helloworld
+  namespace: flux-system
+spec:
+  approvePlan: "auto"
+  path: ./
+  sourceRef:
+    kind: GitRepository
+    name: helloworld
+    namespace: flux-system
+  healthChecks:
+    - name: rds
+      type: tcp
+      url: "{{.rdsUrl}}:{{.rdsPort}}" # uses standard Go package template format to parse outputs to url
```

Comment:
The concept of using `url` is better than `address` 👍 

To make it consistent, `url` would be specified only in one of the following forms:
- `tcp://{{.rdsUrl}}:{{.rdsPort}}`
- `http://...`
- `https://...`
maybe we need a check for this pattern.